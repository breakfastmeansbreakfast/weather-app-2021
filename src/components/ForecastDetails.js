/* eslint-disable react/no-unused-prop-types */
import "../styles/App.css";
import React from "react";
import PropTypes from "proptypes";

function ForecastDetails({ forecast }) {
  const { date, temperature, wind } = forecast;
  return (
    <div className="forecast-details" data-testid="forecast-details">
      <div className="forecast-details__date" data-testid="date-id">
        {new Date(date).toLocaleDateString("en-gb", {
          year: "numeric",
          month: "long",
          day: "numeric",
        })}
      </div>
      <div
        className="forecast-details__temperature-max"
        data-testid="temperature-max-id"
      >
        {temperature.max}&deg;C
      </div>
      <div
        className="forecast-details__temperature-min"
        data-testid="temperature-min-id"
      >
        {temperature.min}&deg;C
      </div>
      <div className="forecast-details__wind-speed" data-testid="wind-speed-id">
        {wind.speed} mph
      </div>
      <div
        className="forecast-details__wind-direction"
        data-testid="wind-direction-id"
      >
        Wind direction: {wind.direction.toUpperCase()}
      </div>
    </div>
  );
}

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number,
    humidity: PropTypes.number,
    temperature: PropTypes.shape({
      min: PropTypes.number,
      max: PropTypes.number,
    }),
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }),
  }).isRequired,
};
export default ForecastDetails;
