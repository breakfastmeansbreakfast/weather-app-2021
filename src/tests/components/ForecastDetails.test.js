import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("it rents ok", () => {
  const validProps = {
    date: 1525392000000,
    temperature: {
      max: 11,
      min: 4,
    },
    wind: {
      speed: 13,
      direction: "s",
    },
    humidity: 30,
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
