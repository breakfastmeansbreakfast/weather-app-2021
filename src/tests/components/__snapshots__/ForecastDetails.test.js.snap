// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it rents ok renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  >
    <div
      class="forecast-details__date"
      data-testid="date-id"
    >
      4 May 2018
    </div>
    <div
      class="forecast-details__temperature-max"
      data-testid="temperature-max-id"
    >
      11°C
    </div>
    <div
      class="forecast-details__temperature-min"
      data-testid="temperature-min-id"
    >
      4°C
    </div>
    <div
      class="forecast-details__wind-speed"
      data-testid="wind-speed-id"
    >
      13 mph
    </div>
    <div
      class="forecast-details__wind-direction"
      data-testid="wind-direction-id"
    >
      Wind direction: S
    </div>
  </div>
</DocumentFragment>
`;
